src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting_NetFx/HtmlFormattingTest.cs

```diff
@@ -440,6 +440,46 @@ @typeparam TValue
         }
     }
 
+    [FormattingTestFact]
+    public async Task PreprocessorDirectives()
+    {
+        await RunFormattingTestAsync(
+            input: """
+            <div Model="SomeModel">
+            <div />
+            @{
+            #if DEBUG
+            }
+             <div />
+            @{
+            #endif
+            }
+            </div>
+
+            @code {
+                private object SomeModel {get;set;}
+            }
+            """,
+            expected: """
+            <div Model="SomeModel">
+                <div />
+                @{
+            #if DEBUG
+                    }
```

Comment:
huh... this looks wrong to me :/

Reply:
I don't understand this either TBH. Is the preprocessor directive increasing the indent level? It doesn't seem to in standalone C# file. E.g.
```
if (true)
{
#if DEBUG
if (true)
{
#endif
}
}
```
formatted becomes
```
if (true)
{
#if DEBUG
    if (true)
    {
#endif
    }
}
```
which is what I would expect

Reply:
To be very clear, this PR doesn't attempt to actually adjust how this document is formatted, it just makes the formatting engine not crash. The `{get;set;}` being spaced out correctly here is providing value to the user that they didn't get before.

Having said that, in the tests there are no preprocessor defines (actually we have no way to pass them to the Razor compiler at all I don't think) so the code in the `#if` is not parsed or formatted by Roslyn. Pretend it doesn't exist, and the result looks right.